src/language/interpreter_new.rs

```diff
@@ -1019,316 +1019,316 @@ impl Sqrt for i64 {
     }
 }
 
+extern crate test;
+
 #[cfg(test)]
 mod tests {
+
     use super::*;
     use approx::AbsDiffEq;
     use ndarray::array;
     use std::str::FromStr;
+    use test::Bencher;
+
+    /// Creates an benchmark test for interpreter
+    /// The test does the following:
+    ///  1. Parses $glenside_str as glenside expression
+    ///  2. Creates a new Environment from the vector of (key, value) pairs in base_env
+    ///  3. Calls check_correct with the glenside expression + environment
+    /// $test_name: the name of the created benchmark test
+    /// $glenside_str: A string containing the Glenside program
+    /// $base_env: A vector or 2-tuples of key, value pairs to put into the environment
+    /// $check_correct: A closure with arguments (glenside_expr, env) that should call the interpreter
+    ///     and checks for correctness
+    macro_rules! benchmark_test {
+        ($test_name:ident, $glenside_str: expr, $base_env: expr, $check_correct: expr) => {
```

Comment:
```suggestion
        ($test_name:ident, $glenside_str: expr, $env: expr, $check_correct: expr) => {
```
Can we just call this `$env`? Is it called `$base_env` for a specific reason?

Reply:
the variable is called `env` here `let mut env = Environment::new();`

I could use a different parameter name if it would make it more clear